server/api.py

```diff
@@ -86,10 +87,17 @@ def login(self, params):
         payload = {
             'nonce': nonce,
             'email': user['email'],
-            'external_id': user['_id'],
+            'external_id': str(user['_id']),
             'username': user['login'],
             'name': '%s %s' % (user['firstName'], user['lastName']),
-            'require_activation': 'true' if requireActivation else 'false'
+            'require_activation': 'true' if requireActivation else 'false',
+            'admin': 'true' if user['admin'] else 'false',
+            'add_groups': ','.join(
+                group['name']
```

Comment:
Does the group name have to match the "Name" group field on Discourse as opposed to the "Full Name" field? If so, it looks like there will be a mismatch with the current configuration.

See https://github.com/discourse/discourse/blob/dd101e4f1dc556ba1fcfde5e95c39665bff9792d/app/models/discourse_single_sign_on.rb#L113.